File: src/backend/Intelibill.Domain/Entities/BankAccount.cs
using Intelibill.Domain.Common;
using Intelibill.Domain.Enums;

namespace Intelibill.Domain.Entities;

public sealed class BankAccount : BaseEntity
{
    public Guid ShopId { get; private set; }
    public string BankName { get; private set; } = string.Empty;
    public string AccountNumber { get; private set; } = string.Empty;
    public BankAccountType? AccountType { get; private set; }
    public string? IfscCode { get; private set; }
    public string? AccountHolderName { get; private set; }

    private BankAccount() { }

    public static BankAccount Create(
        Guid shopId,
        string bankName,
        string accountNumber,
        BankAccountType? accountType,
        string? ifscCode,
        string? accountHolderName)
    {
        return new BankAccount
        {
            ShopId = shopId,
            BankName = bankName.Trim(),
            AccountNumber = accountNumber.Trim(),
            AccountType = accountType,
            IfscCode = ifscCode?.Trim(),
            AccountHolderName = accountHolderName?.Trim()
        };
    }

    public void Update(
        string bankName,
        string accountNumber,
        BankAccountType? accountType,
        string? ifscCode,
        string? accountHolderName)
    {
        BankName = bankName.Trim();
        AccountNumber = accountNumber.Trim();
        AccountType = accountType;
        IfscCode = ifscCode?.Trim();
        AccountHolderName = accountHolderName?.Trim();
    }
}

File: src/mobile/android/intelibill_mobile/lib/src/features/bank_accounts/domain/entities/bank_account.dart
import 'package:equatable/equatable.dart';

class BankAccount extends Equatable {
  const BankAccount({
    required this.id,
    required this.bankName,
    required this.accountNumber,
    this.accountType,
    this.ifscCode,
    this.accountHolderName,
  });

  final String id;
  final String bankName;
  final String accountNumber;
  final String? accountType;
  final String? ifscCode;
  final String? accountHolderName;

  @override
  List<Object?> get props => [
    id,
    bankName,
    accountNumber,
    accountType,
    ifscCode,
    accountHolderName,
  ];
}
